apps/scan/frontend/src/screens/poll_worker_screen.tsx

```diff
@@ -106,17 +278,18 @@ const ButtonGrid = styled.div`
 `;
 
 export function PollWorkerScreen({
+  initialPollsInfo,
```

Comment:
What's the motivation for passing in an initial polls info and then also querying polls info within the component?

Reply:
Because I want to set the initial component state (`pollWorkerFlowState`). Setting it after the query completes would force me to use an effect, I thought. Do you have an idea for a better pattern for this? I agree it's weird.

Reply:
My go-to pattern is to use a parent component that has the query and renders a loading state until the data is loaded. Then the child component has a required prop for the result of the query. (Which is basically what was already set up here, with `AppRoot` as the parent).

Sometimes it's nice for testing to have a screen component manage all its own queries though, in which case I just make the screen component the parent and add a child as needed. E.g.:

```ts
function PollWorkerScreen() {
  const pollsInfoQuery = getPollsInfo.useQuery()
  if (!pollsInfoQuery.isSuccess) return null
  return <PollWorkerScreenContents pollsInfo={pollsInfo} />
}
```
}

Reply:
I added a wrapper in my latest commit, but it brought me back to why I originally did it in this weird way.

Do you remember this discussion? https://github.com/votingworks/vxsuite/pull/4222#discussion_r1396218990

Some states of the poll worker screen depend on the `pollsInfo` being up to date. E.g., by the time we go to print a report, we expect `pollsInfo` to be updated with the latest polls transition. We can use `async` `onSuccess` invalidations to guarantee this, but not if the mutation and the query are in different components, which is what happens when we use the suggested pattern. 

To address this, using a wrapper, I added that state that we were asserting exists into the `FlowState`.

62ea498fb